src/shims/fs.rs

```diff
@@ -1103,6 +1113,89 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.handle_not_found()
         }
     }
+
+    fn fsync(&mut self, fd_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        // On macOS, `fsync` (unlike `fcntl(F_FULLFSYNC)`) does not wait for the
+        // underlying disk to finish writing. In the interest of host compatibility,
+        // we conservatively implement this with `sync_all`, which
+        // *does* wait for the disk.
+
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("fsync")?;
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+        if let Some(FileHandle { file, writable }) = this.machine.file_handler.handles.get_mut(&fd) {
+            if !*writable && cfg!(windows) {
+                // sync_all() will return an error on Windows hosts if the file is not opened for writing.
```

Comment:
Is there some docs or something we can show at that documents this as correct/expected behavior?

If not, please add a FIXME that we need to verify if this is correct.